### DISCUSSION TITLE
Ability to customize Counter look

### DISCUSSION THREAD

**Discussion by cezudas** [getredash/redash#7176] [Q&A]
Hi, I was wondering if there is any means to eachieve a small degree of customization for the counters, like the ability to specify the font color
![Screenshot 2024-09-25 at 09 24 32](https://github.com/user-attachments/assets/98fea458-63f5-4f4e-9e6d-954307242c36)

I'm a new Redash user and I'm missing most of the context. I think the fact that Counter widget doesn't support HTML has been mentioned a few times by the maintainers, for example [here](https://discuss.redash.io/t/conditional-formatting-general-text-formatting/1706).
I'd be happy to contribute with a PR, I just wanted to hear out what our maintainers think about it.

**Comment by deed02392** [comment]
Pinging @eradman – maybe current maintainers have no opinion about it. At the very least if not HTML support, we'd like to be able to make the counter support a custom hyperlink.

**Reply by eradman** [reply]
The ability to customize the counter widget would be useful. I haven't given it much thought, but HTML support seems reasonable.

**Reply by deed02392** [reply]
comparing with the Table viz, it seems that Counter could be rationalized. note in Table you can use Display as: Number and specify formatting, whereas in Counter it has a dedicated tab. we could give Counter a Display as: Text type with optional HTML support, Number which enables the same features under Format, and Link for Redash-style hyperlink configuration. It seems that if Redash doesn't detect a number in the data, all the fields under the Format tab are disabled anyway.

**Reply by eradman** [reply]
Also, instead of

Formatting Decimal Place
Formatting Decimal Character
Formatting Thousands Separator
Formatting String Prefix
Formatting String Suffix

We should just support a _Number format_ the same way that Table does!

**Reply by deed02392** [reply]
Yes exactly, these could all be rolled up under a `Display as: Number` feature. I imagine Table evolved whereas was left in its more primitive, original design. Or someone decided it should only ever display a number and therefore breaking out the formatting makes sense.
